sdk/src/metrics/data/circular_buffer.cc

```diff
@@ -83,13 +83,15 @@ struct AdaptingIntegerArrayCopy
 
 void AdaptingIntegerArray::Increment(size_t index, uint64_t count)
 {
-  const uint64_t result = nostd::visit(AdaptingIntegerArrayIncrement{index, count}, backing_);
-  if OPENTELEMETRY_LIKELY_CONDITION (result == 0)
+  while (true)
```

Comment:
Please explain in more details the change.

Some comments in the code are needed, very hard to understand what it actually does.

Why a while(true) loop, and is there a risk it never ends ?

Reply:
It's avoid the recursive call. Instead makes an Iterative while loop. 

Anyway reverting this back. Will add in next small PR.

Reply:
Thanks. Fixing all the trivial issues first will reduce the overall noise, so it will be easier to address remaining (and less trivial) fixes later.